Document official Datadog OpenCode plugin for MCP Server (#36624)
DataDog/documentation: content/en/bits_ai/mcp_server/setup.md

{{% tab "OpenCode" %}}

Connect [OpenCode][3] to the Datadog MCP Server with the official [Datadog OpenCode Plugin][2] (in Preview). The plugin writes and maintains the MCP Server entry in your `opencode.json` and exposes the `ddsetup`, `ddconfig`, and `ddtoolsets` tools that the agent uses to handle setup, site changes, and [toolset](#toolsets) selection.

{{< site-region region="us,us3,us5,eu,ap1,ap2" >}}

1. Add the plugin to your `opencode.json` configuration file. Create the file if it doesn't exist:

   <pre><code>{
     "plugin": ["@datadog/opencode-plugin"]
   }</code></pre>

>     If your `opencode.json` already has a `plugin` array, append `"@datadog/opencode-plugin"` to it.

Reviewer: This detail seems a little unnecessary for our typical audience, who probably know how to update JSON entries. But fine to keep if it's something that we know customers have needed help with.

    If you previously configured the Datadog MCP Server manually in `opencode.json`, remove or disable that entry to avoid conflicts with the plugin.

> 1. Restart OpenCode. The package is fetched from npm at startup, so no separate install step is needed.

Reviewer: ```suggestion
1. Restart OpenCode. The package is fetched from npm at startup.
```

> 1. Ask the agent a Datadog-related question, or tell it to run `ddsetup`. The plugin walks through site selection and writes the MCP Server configuration to `opencode.json`.

Reviewer: If the agent is going to walk through site selection either way, we should probably just tell users to run `ddsetup`. It could be annoying if we tell users they can start asking questions only to have them get interrupted by more setup. 

Also, we already mentioned in the intro that the plugin writes to `opencode.json`, no need to repeat this.
```suggestion
1. Ask the agent to run `ddsetup`. The plugin walks through site selection.
```

1. Restart OpenCode again to activate the MCP Server, and complete the OAuth login flow when prompted.

1. Verify that you have the required [permissions](#required-permissions) for the Datadog resources you want to access.

> After setup, ask the agent to run `ddconfig` to change your Datadog site or troubleshoot the connection, or `ddtoolsets` to enable or disable groups of [tools](#toolsets).

Reviewer: Enabling tools should be an explicit setup step, as this can really affect customer experience if they don't have the right tools enabled for the Datadog features they want to interact with. Running `ddconfig` seems more like a post-install optional thing.
```suggestion
1. To enable [product-specific tools](#toolsets), ask the agent to run `ddtoolsets`.

After setup, ask the agent to run `ddconfig` to change your Datadog site or troubleshoot the connection.
```

{{% collapse-content title="Manual configuration" level="h4" expanded=false id="opencode-manual" %}}

> To configure the MCP Server without the plugin, add the following to your `opencode.json` configuration file. Selected endpoint ({{< region-param key="dd_site_name" >}}): <code>{{< region-param key="mcp_server_endpoint" >}}</code>.

Reviewer: ```suggestion
To configure the MCP Server without the plugin, add the following to your `opencode.json` configuration file. 

Selected endpoint ({{< region-param key="dd_site_name" >}}): <code>{{< region-param key="mcp_server_endpoint" >}}</code>.
```
